### PR TITLE
Cleanup missiles

### PR DESCRIPTION
**Pull request by AJenbo**
About 1/3 of the changes are simply indentation so might want to play with the "ignore white space" option, also i split the most intrusive changes in to a second commit.

### CODE REVIEW

File: Source/missiles.cpp [hunk 1]
```diff
@@ -562,7 +552,7 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, int t, bool
 	if (monster[m]._mmode == MM_CHARGE)
 		return false;
 
-	mor = monster[m].mMagicRes;
+	uint8_t mor = monster[m].mMagicRes;
```
Comment (StephenCWills) [inline Source/missiles.cpp:555]: Isn't `mMagicRes` a uint16_t?

File: Source/missiles.cpp [hunk 2]
```diff
@@ -1552,7 +1545,7 @@ void AddStealPotions(int mi, Point src, Point dst, int midir, int8_t mienemy, in
 					SetPlrHandItem(&player.HoldItem, ii);
 					GetPlrHandSeed(&player.HoldItem);
 					player.HoldItem._iStatFlag = true;
-					player.SpdList[si] = plr[pnum].HoldItem;
+					player.SpdList[si] = player.HoldItem;
```
Comment (StephenCWills) [inline Source/missiles.cpp:1548]: Nice catch. 👍

**Review by StephenCWills — CHANGES_REQUESTED** [review]
I only had the one question about `mMagicRes`. Everything else looks good to me.

File: Source/missiles.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/missiles.cpp:555]:
```
MonsterStruct: uint16_t
UniqMonstStruct: uint16_t
MonsterData: uint8_t
```

It represents `monster_resistance` (`uint8_t`) as a bitflag.

**Review by ephphatha — APPROVED** [review]

**Review by StephenCWills — APPROVED** [review]

**Pull request merged**